File: DSA/Raima_Rekhi_590013023/Day16Question1.c
#include <stdio.h>

int linear_search(int arr[],int n,int element);

int main(){
int n,element;

printf("Enter the size of the array:\n");
scanf("%d",&n);
int arr[n];

// Prompt the user to enter the elements of the array
printf("Enter the elements of the array:\n");
for(int i=0;i<n;++i){
scanf("%d",&arr[i]);
}

// Prompt the user to enter the element to be searched in the array
printf("Enter the target element:\n");
scanf("%d",&element);

int index=linear_search(arr,n,element);

if(index!=-1)
printf("The target element %d found at index: %d\n",element,index);
else
printf("%d\nNo such element found.\n",index);
return 0;
}

// Function to find the index of an element in an array using linear search technique
int linear_search(int arr[],int n,int element){
for(int i=0;i<n;++i){
if(arr[i]==element)
return i;
}
return -1;
}

File: DSA/Krish_Mishra_590015053/Day16_Question1.py
def linear_search(array, target):
    for index in range(len(array)):
        if array[index] == target:
            return index
    return -1
user_input = input("Enter the elements of the array separated by spaces: ")
array = list(map(int, user_input.split()))
target = int(input("Enter the target element to search: "))
result = linear_search(array, target)
if result != -1:
    print(f"Target found at index {result}")
else:
    print("Target not found in the array.")
